File: MobileChallenge/MobileChallenge/UIImageView+DownloadImage.m
//
//  UIImageView+DownloadImage.m
//  MobileChallenge
//
//  Created by Praneet Jain on 9/15/16.
//  Copyright © 2016 Praneet Jain. All rights reserved.
//

#import "UIImageView+DownloadImage.h"

@implementation UIImageView (DownloadImage)

//This method downloads image from url in background. Once downloaded, it updates the imageview for album.
-(NSURLSessionDownloadTask*)loadImageWithURL:(NSURL*)url{

    NSURLSession* session =[NSURLSession sharedSession];
    
    NSURLSessionDownloadTask* downloadTask = [session downloadTaskWithURL:url completionHandler:^(NSURL* location, NSURLResponse* response, NSError* error) {
        
        if(error==nil){
        
            NSData* data = [NSData dataWithContentsOfURL:url];
            if(data!=nil){
            UIImage* image = [UIImage imageWithData:data];
            
            dispatch_async(dispatch_get_main_queue(), ^{
                self.image = image;
            });
            }
        }
        
    }];
    
    [downloadTask resume];
    
    return downloadTask;
    
    
  }

@end

File: StoreSearch/StoreSearch/UIImageView+DownloadImage.swift
//
//  UIImageView+DownloadImage.swift
//  StoreSearch
//
//  Created by Praneet Jain on 5/23/16.
//  Copyright © 2016 Praneet Jain. All rights reserved.
//

import Foundation
import UIKit

extension UIImageView{
    func loadImageWithURL(url: NSURL) -> NSURLSessionDownloadTask {
        let session = NSURLSession.sharedSession()
        //1
        let downloadTask = session.downloadTaskWithURL(
            url, completionHandler: { [weak self] url, response, error in
                // 2
                if error == nil, let url = url,
                    // 3
                    data = NSData(contentsOfURL: url), image = UIImage(data: data) {
                    // 4
                    dispatch_async(dispatch_get_main_queue()) {
                        if let strongSelf = self {
                            strongSelf.image = image
                        }
                    }
                }
            })
        // 5
        downloadTask.resume()
        return downloadTask
    }
}

